packages/core/src/2d/sprite/SpriteRenderer.ts

```diff
@@ -154,9 +160,9 @@ export class SpriteRenderer extends Renderer {
     const { transform } = this.entity;
 
     // Update sprite data.
-    const localDirty = sprite._updateMeshData();
+    sprite._updateMeshData();
 
-    if (this._isWorldMatrixDirty.flag || localDirty || this._isContainDirtyFlag(DirtyFlag.Sprite)) {
+    if (this._isWorldMatrixDirty.flag || this._spriteDirty.flag || this._isContainDirtyFlag(DirtyFlag.Sprite)) {
```

Comment:
If we use `this._spriteDirty.flag`, `this._isContainDirtyFlag(DirtyFlag.Sprite)` is no need!